pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -390,14 +392,22 @@ def append_fn(v: ResultMetric) -> None:
         apply_to_collection(list(self.values()), ResultMetric, append_fn)
         return o
 
-    @property
-    def batch_size(self) -> torch.Tensor:
-        # performance: cache the `batch_size` tensor instead of re-creating it
-        return self._batch_size
-
-    @batch_size.setter
-    def batch_size(self, value: int) -> None:
-        self._batch_size = torch.tensor(value, device=self.device)
+    def _extract_batch_size(self, batch_size: Optional[int], meta: _Metadata) -> int:
+        # check if we have extracted the batch size already
+        batch_size = batch_size or self.batch_size
+        if batch_size is not None:
+            return batch_size
+        batch_size = 1
+
+        # extract it
+        fx_validate = _FxValidator.functions.get(meta.fx.split(".")[0])
+        if meta.on_epoch and fx_validate is not None and (True in fx_validate["on_step"]) and meta.is_mean_reduction:
+            try:
+                batch_size = extract_batch_size(self.batch)
+                self.batch_size = batch_size
+            except RecursionError:
```

Comment:
@rohitgr7  There could be other exceptions coming from `extract_batch_size()`. For example, if batch is ```
batch={'a':torch.tensor([1, 2, 3, 4, 5]), 'b': [torch.tensor(1), torch.tensor(2)]}
```
it raises index error when batch.size(0) is called for torch.tensor(1). 
I understand the good intention to automatically retrieve batch_size from batch, but we can't control how user set up their batch input. It's better to involve  user to provide batch_size.

Reply:
good catch! I guess this is the only edge case we had with tensors. Added a check for it.